src/vdaf/prio3.rs

```diff
@@ -105,57 +105,43 @@ impl Prio3Aes128CountVecMultithreaded {
         num_aggregators: u8,
         len: usize,
     ) -> Result<Self, VdafError> {
-        check_num_aggregators(num_aggregators)?;
-
-        Ok(Prio3 {
-            num_aggregators,
-            typ: CountVec::new(len),
-            phantom: PhantomData,
-        })
+        Prio3::new(num_aggregators, CountVec::new(len))
     }
 }
 
 /// The sum type. Each measurement is an integer in `[0,2^bits)` for some `0 < bits < 64` and the
 /// aggregate is the sum.
+#[cfg(feature = "internal_prg")]
 pub type Prio3Aes128Sum = Prio3<Sum<Field128>, PrgAes128, 16>;
 
+#[cfg(feature = "internal_prg")]
 impl Prio3Aes128Sum {
     /// Construct an instance of Prio3Aes128Sum with the given number of aggregators and required
     /// bit length. The bit length must not exceed 64.
     pub fn new_aes128_sum(num_aggregators: u8, bits: u32) -> Result<Self, VdafError> {
-        check_num_aggregators(num_aggregators)?;
-
         if bits > 64 {
             return Err(VdafError::Uncategorized(format!(
                 "bit length ({}) exceeds limit for aggregate type (64)",
                 bits
             )));
         }
 
-        Ok(Prio3 {
-            num_aggregators,
-            typ: Sum::new(bits as usize)?,
-            phantom: PhantomData,
-        })
+        Prio3::new(num_aggregators, Sum::new(bits as usize)?)
     }
 }
 /// The histogram type. Each measurement is an unsigned integer and the result is a histogram
 /// representation of the distribution. The bucket boundaries are fixed in advance.
+#[cfg(feature = "internal_prg")]
 pub type Prio3Aes128Histogram = Prio3<Histogram<Field128>, PrgAes128, 16>;
 
+#[cfg(feature = "internal_prg")]
 impl Prio3Aes128Histogram {
     /// Constructs an instance of Prio3Aes128Histogram with the given number of aggregators and
     /// desired histogram bucket boundaries.
     pub fn new_aes128_histogram(num_aggregators: u8, buckets: &[u64]) -> Result<Self, VdafError> {
-        check_num_aggregators(num_aggregators)?;
-
         let buckets = buckets.iter().map(|bucket| *bucket as u128).collect();
 
-        Ok(Prio3 {
-            num_aggregators,
-            typ: Histogram::<Field128>::new(buckets)?,
-            phantom: PhantomData,
-        })
+        Prio3::new(num_aggregators, Histogram::<Field128>::new(buckets)?)
```

Comment:
If possible (i.e., rustc is happy):
```suggestion
        Prio3::new(num_aggregators, Histogram::new(buckets)?)
```